core/src/test/java/com/linecorp/armeria/client/endpoint/healthcheck/HttpHealthCheckedEndpointGroupTest.java

```diff
@@ -75,13 +76,15 @@ public void endpoints() throws Exception {
                 new StaticEndpointGroup(
                         Endpoint.of("127.0.0.1", 1234),
                         Endpoint.of("127.0.0.1", 2345)),
-                metricRegistry,
                 healthCheckPath);
+        metricRegistry.registerAll(endpointGroup.metricSet("metric"));
         assertThat(endpointGroup.endpoints()).isEmpty();
-        assertThat(metricRegistry.getGauges().get("health-check.127.0.0.1:1234.healthy").getValue()).isEqualTo(
-                0);
-        assertThat(metricRegistry.getGauges().get("health-check.127.0.0.1:2345.healthy").getValue()).isEqualTo(
-                0);
+        assertThat(metricRegistry.getGauges().get("healthcheck.metric.all.server.count").getValue())
+                .isEqualTo(2);
+        assertThat(metricRegistry.getGauges().get("healthcheck.metric.healthy.server.count").getValue())
+                .isEqualTo(0);
+        assertThat(metricRegistry.getGauges().get("healthcheck.metric.healthy.servers").getValue())
+                .isEqualTo(ImmutableSet.of());
 
         ServiceServer serverOne = new ServiceServer(healthCheckPath, 1234).start();
         ServiceServer serverTwo = new ServiceServer(healthCheckPath, 2345).start();
```

Comment:
You need to specify the port number `0` and get the actual port number after the server is started to avoid port conflicts.